dj/sql/parsing/ast.py

```diff
@@ -244,47 +244,68 @@ def __str__(self) -> str:
             f"{self.quote_style}{self.name}{self.quote_style}"  # pylint: disable=C0301
         )
 
-    def to_identifier(self) -> "Identifier":
+    def to_namespace(self) -> "Namespace":
         """
         transforms a single Name to a single item Identifier
         """
-        return cast(Identifier, Identifier([self]).add_self_as_parent())
+        return cast(Namespace, Namespace([self]).add_self_as_parent())
 
     def __hash__(self) -> int:
         return hash(self.name + self.quote_style)
 
 
 @dataclass(eq=False)
-class Identifier(Node):
+class Namespace(Node):
```

Comment:
This class implies that we'd go `Namespace -> Column` or `Namespace -> Table`, are there implications if we design it to go the other way and logically guarantee that a `Namespace` instance never includes the last item (column name or table name)? Something along the lines of:
```py
>>> t = Table("n1.n2.n3.n4.tbl1")
>>> t.name
"tbl1"
>>> t.namespace
Namespace(
    names=[
        Name(name="n1", quote_style=""),
        Name(name="n2", quote_style=""),
        Name(name="n3", quote_style=""),
        Name(name="n4", quote_style=""),
    ]
)